### PR TITLE
Ordering Questions while creating Question Paper

### PR DESCRIPTION
**Pull request by adityacp**
This PR will add questions to Question Paper in the order of selection.
The Questions while displaying will not order by key and will show up according to selections made during creation of question paper.
This fixes issue #213

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -748,14 +757,33 @@ def create_demo_quiz_ppr(self, demo_quiz, user):
...
     def __str__(self):
         return "Question Paper for " + self.quiz.description
 
 
 ###############################################################################
+class FixedQuestions(models.Model):
+    questionpaper = models.ForeignKey(QuestionPaper)
+    question = models.ForeignKey(Question)
+
+    def get_fixed_questions(self, qp):
```
Comment (maheshgudi) [inline yaksh/models.py]: I think, it would be better to make this a model manager method.

File: yaksh/models.py [hunk 2]
```diff
@@ -748,14 +757,33 @@ def create_demo_quiz_ppr(self, demo_quiz, user):
...
+        fixed_q = FixedQuestions.objects.filter(questionpaper=qp).order_by('pk')
+        questions = [que.question for que in fixed_q
+            if que.question.active == True]
+        return questions
+
+    def add_fixed_questions(self, qp, que):
```
Comment (maheshgudi) [inline yaksh/models.py]: same here

File: yaksh/models.py [hunk 3]
```diff
@@ -748,14 +757,33 @@ def create_demo_quiz_ppr(self, demo_quiz, user):
...
+        FixedQuestions.objects.create(questionpaper=qp, question=que)
+
+    def remove_fixed_questions(self, que_ids):
```
Comment (maheshgudi) [inline yaksh/models.py]: same here

File: yaksh/models.py [hunk 4]
```diff
@@ -619,8 +619,8 @@ class QuestionPaperManager(models.Manager):
     def _create_trial_from_questionpaper(self, original_quiz_id):
         """Creates a copy of the original questionpaper"""
         trial_questionpaper = self.get(quiz_id=original_quiz_id)
-        trial_questions = {"fixed_questions": trial_questionpaper
-                           .fixed_questions.all(),
+        fixed_ques = trial_questionpaper.get_ordered_questions()
```
Comment (prathamesh920) [inline yaksh/models.py]: Fine for now, but can have more readable variable name.

**Pull request merged**